Formula/burst.rb

```diff
@@ -0,0 +1,64 @@
+class Burst < Formula
+  desc "Radix sort, lazy ranges and iterators, and more. Boost-like header-only library"
+  homepage "https://github.com/izvolov/burst"
+  url "https://github.com/izvolov/burst/archive/refs/tags/v3.1.1.tar.gz"
+  sha256 "ee58c7b98ca1709dd452b9ba46cb4c91fc0b2952edd020ed5bc2d600b3edeae7"
+  license "BSL-1.0"
+  head "https://github.com/izvolov/burst.git", branch: "master"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "doctest" => :build
+  depends_on "boost"
+
+  on_linux do
+    depends_on "gcc@9"
+  end
+
+  fails_with :gcc do
+    version "6"
+    cause "Requires C++14 constexpr"
+  end
+  fails_with gcc: "12"
```

Comment:
```suggestion
  end

  fails_with gcc: "12"
```
No `cause` for `gcc-12`?

Reply:
After some hours of research I'm pretty sure that cause is optimization bug in GCC12.{0,1}. But I can't prove it strictly.